compiler/p/codegen/PPCBinaryEncoding.cpp

```diff
@@ -268,49 +296,13 @@ uint8_t *TR::PPCConditionalBranchInstruction::generateBinaryEncoding()
       {
       if (label->getCodeLocation() != NULL)
          {
-         if (!getFarRelocation())
-            {
-            insertConditionRegister((uint32_t *)cursor);
-            *(int32_t *)cursor |= ((label->getCodeLocation()-cursor) & 0xffff);
-            }
-         else // too far - need fix up
-            {
-            TR::InstOpCode::Mnemonic reversedOp;
-            bool  linkForm = reversedConditionalBranchOpCode(getOpCodeValue(), &reversedOp);
-            TR::InstOpCode reversedOpCode(reversedOp);
-            TR::InstOpCode extraOpCode(linkForm?TR::InstOpCode::bl:TR::InstOpCode::b);
-
-            cursor = reversedOpCode.copyBinaryToBuffer(cursor);
-            insertConditionRegister((uint32_t *)cursor);
-            *(int32_t *)cursor |= 0x0008;
-            cursor += 4;
-
-            cursor = extraOpCode.copyBinaryToBuffer(cursor);
-            *(int32_t *)cursor |= ((label->getCodeLocation() - cursor) & 0x03fffffc);
-            }
+         insertConditionRegister((uint32_t *)cursor);
+         *(int32_t *)cursor |= ((label->getCodeLocation()-cursor) & 0xffff);
          }
       else
          {
-         if (!getFarRelocation())
-            {
-            insertConditionRegister((uint32_t *)cursor);
-            cg()->addRelocation(new (cg()->trHeapMemory()) TR::LabelRelative16BitRelocation(cursor, label));
-            }
-         else // too far - need fix up
-            {
-            TR::InstOpCode::Mnemonic reversedOp;
-            bool  linkForm = reversedConditionalBranchOpCode(getOpCodeValue(), &reversedOp);
-            TR::InstOpCode reversedOpCode(reversedOp);
-            TR::InstOpCode extraOpCode(linkForm?TR::InstOpCode::bl:TR::InstOpCode::b);
-
-            cursor = reversedOpCode.copyBinaryToBuffer(cursor);
-            insertConditionRegister((uint32_t *)cursor);
-            *(int32_t *)cursor |= 0x0008;
-            cursor += 4;
-
-            cursor = extraOpCode.copyBinaryToBuffer(cursor);
-            cg()->addRelocation(new (cg()->trHeapMemory()) TR::LabelRelative24BitRelocation(cursor, label));
-            }
+         insertConditionRegister((uint32_t *)cursor);
```

Comment:
This line doesn't need to be on both sides of the if/else.

Reply:
Fixed in 0fbf852bc2c3f1b74f8d67f95138e3d15ea09dfb